src/main/java/boss/Parser.java

```diff
@@ -0,0 +1,129 @@
+package boss;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents the class that deals with
+ * making sense of the user command
+ */
+
+public class Parser {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+
+    /**
+     * Creates a Parser object!
+     * @param storage deals with loading tasks and saving tasks from file
+     * @param tasks contains the task list
+     */
+    public Parser(Storage storage, TaskList tasks, Ui ui) {
+        this.storage = storage;
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    /**
+     * Handles the user command
+     * @param task the user command
+     * @throws IOException throws an exception if
+     *                     writing to file causes an error.
+     */
+
+    public void handleCommand(String task) throws IOException, BossException {
+        if (task.equals("list")) {
+            storage.printFileContents();
+        } else if (task.startsWith("unmark")) {
+            String newFileData = tasks.unmark(task);
+            storage.writeToFile(newFileData, false);
+        } else if (task.startsWith("mark")) {
+            String newFileData = tasks.mark(task);
+            storage.writeToFile(newFileData, false);
+        } else if (task.startsWith("delete")) {
+            String newFileData = tasks.delete(task);
+            storage.writeToFile(newFileData, false);
+        } else if (task.startsWith("find")) {
+            String description = task.split("find ")[1];
+            tasks.find(description);
+        }
+        // it must be a command to create a task (of some type)
+        else {
+            Boss.Types taskType;
+            if (task.startsWith("todo")) {
+                taskType = Boss.Types.TODO;
+            } else if (task.startsWith("deadline")) {
+                taskType = Boss.Types.DEADLINE;
+            } else if (task.startsWith("event")) {
+                taskType = Boss.Types.EVENT;
+            } else {
+                taskType = Boss.Types.NONE;
+            }
+
+            switch (taskType) {
+            case TODO:
+                String[] string = task.split(" ");
+                if (string.length == 1) {
+                    ui.showLoadingError(new BossException("The description of todo cannot be empty!"));
```

Comment:
An interesting way to implement different error messages! Could it be made neater by creating a few set error messages?